lib/config/presets/index.ts

```diff
@@ -340,15 +363,67 @@ export async function resolveConfigPresets(
         val as RenovateConfig,
         baseConfig,
         ignorePresets,
-        existingPresets
+        existingPresets,
+        shallowResolve
       );
     }
   }
   logger.trace({ config: inputConfig }, 'Input config');
   logger.trace({ config }, 'Resolved config');
+  if (shallowResolve) {
+    handleExtendsArray(config, unresolvedPresets);
+  }
   return config;
 }
 
+// handleExtendsArray:
+// This function edits extends array presets when shallowResolve is enabled
+// 1. remove resolved external presets from the extends array.
+// 2. clean duplicate presets keys (from different resolution levels).
+function handleExtendsArray(
+  config: AllConfig,
+  unresolvedPresets: string[]
+): void {
+  if (!config.extends?.length) {
+    return;
+  }
+  const uniqueExtends = new Set([
+    ...config.extends.filter((e) => !shouldShallowResolve(e)),
```

Comment:
In what scenario would the values in config.extends not already be in unresolvedPresets?

Reply:
My concern here was is when `mergeChildConfig` function was being called its was sometimes removing values from the extends array and that was messing up the log result. So to deal with this I decided to save the values before the calling `mergeChildConfig`.

Reply:
And also to note that another purpose of this function is to clean duplicate presets keys incase they are coming from different resolution levels.